### DISCUSSION TITLE
Is it possible to run multiple instances of FigureResampler?

### DISCUSSION THREAD

**Discussion by dbdq** [predict-idlab/plotly-resampler#306] [Q&A] [answered]
Hi, thanks for writing a great package.

I'm trying to show two figures on two tabs of the browser so that I can compare two figures side by side with resampling function. However, as FigureResampler occupies the main thread, I cannot run both instances at the same time.
I can write two codes that plots different data, but this is obviously not scalable.

What would be the best way to achieve my goal? hanks!

**Comment by jonasvdd** [comment, marked as answer]
Hi @dbdq,

Thank you for your kind words. 

I am missing some context in your question on how you utilize plotly-resampler. Given that you are not able to spawn multiple figures, I suppose that you utilize the *automatic* (i.e., the `register_plotly_resampler`) mode.

However, It is certainly possible to run multiple plotly-resampler figures within a same dash-app or notebook instance. 
- To achieve this with a **dash app**: see [this example](https://github.com/predict-idlab/plotly-resampler/blob/629bab623a16d1497dbae0485ad4ed16acef7a97/examples/dash_apps/03_minimal_cache_dynamic.py)
- To achieve this in a **notebook**: you need to utilize the *manual* mode; and wrap the  `go.Figure` with a `FigureResampler` and call `show_dash` for which you pass the network port on which the app runs as an argument. 
```python
import plotly.graph_objects as go
from plotly_resampler import FigureResampler
import numpy as np

N = 1_000_000
x = np.arange(N)
y = np.sin(x / (N / 50))

fig1 = FigureResampler(go.Figure())
fig1.add_trace(go.Scatter(name='a trace'), hf_x=x, hf_y=y)
# NOTE: via manual mode - you can specify the port on which each figure can run
fig1.show_dash(mode='external', port=8001)

fig2 = FigureResampler(go.Figure())
fig2.add_trace(go.Scatter(name='a trace'), hf_x=x, hf_y=-y)  # y is flipped
fig2.show_dash(mode='external', port=8002)
```


![image](https://github.com/predict-idlab/plotly-resampler/assets/38005924/2bcd1b31-393e-4bdf-b7c1-14467ca956a8)


I hope this resolves your issue.

Kind regards,
Jonas

**Reply by dbdq** [reply]
Thank you very much for the detailed answer.
I was using the manual mode using the same `fig.show_dash(mode='external', port=8001)` syntax but didn't know it only works in notebook if I want to run multiple show_dash() in parallel.
As I was getting a warning that suggests me to use WSGI server instead of Flask, I thought it was the issue.

Is there any way to make it work using the simple `fig.show_dash()` syntax, where it automatically spawns a child thread and runs a server inside it?
